Add example shapes following our standard.

irm-codebase opened the pull request:
Adds four geoparquet files which can be used by modules for testing.
All of these have the standard columns (`shape_id`, `country_id`, `shape_spec`).

- PRT_admin1: multi-region shapefile of portugal (for lightweight testing).
- MNE_admin0: single-region shapefile of Montenegro (for very lightweight testing).
- EU+: sector coupled euro calliope countries.
- ehighways: sector coupled euro calliope ehighways resolution.

brynpickering commented on resources/shapes/MNE_admin0.parquet:
mne_admin0

brynpickering commented on resources/shapes/PRT_admin1.parquet:
prt_admin1

brynpickering commented on resources/shapes/ehighways.parquet:
europe_ehighways

brynpickering commented on resources/shapes/eu+.parquet:
europe_admin0

brynpickering replied:
This one I'm most keen on having the name changed as a `+` in the filename just feels wrong :P

irm-codebase replied:
Removed (for now) until we find a solution to the mismatches.

brynpickering reviewed (changes requested):
I think having the data in Clio is fine (for now) - few people are cloning this repo. I would always squash and merge data PRs to reduce the total burden of files in history.

I've made suggestions for filenames, to keep to snakecase and to go for the convention of <scope>_<resolution> across all files.

I would also reduce the filesizes by:
1. using `compression` on storing to file (in this case, perhaps `compression="gzip"`)
1. [simplifying the geometries](https://gis.stackexchange.com/a/420906). Note that I'm not suggesting using the geopandas simplify, as it doesn't do a good job at borders. But testing this method shows good results on preserving borders and could allow you timplify to a tolerance of 0.05.

With both applied for the `eu+` file, the filesize goes down from 3MB to 326KB. We could then set the filesize limit to 1000KB?

EDIT: final point; could we add to the file metadata with some additional info incl. attributions (as is technically required for NUTS data)? I'm not entirely clear on how it's done via geopandas as it looks to require accessing the [pyarrow ParquetWriter](https://arrow.apache.org/docs/python/generated/pyarrow.parquet.ParquetWriter.html) `add_key_value_metadata` method.

brynpickering commented:
p.s. on testing out geometry simplification, I noticed that the data currently uploaded has some border mismatches, see below.

This will be an issue when using borders to do things like auto-define transmission links. It may be a gadm/nuts combo issue.

## Croatian border
<img width="469" alt="image" src="https://github.com/user-attachments/assets/fa399483-886a-4809-9a11-e9c86b17012c" />

## Bulgarian and Greek borders
<img width="469" alt="image" src="https://github.com/user-attachments/assets/03786f95-d73d-46e6-9920-508c48c7a672" />

irm-codebase commented:
> p.s. on testing out geometry simplification, I noticed that the data currently uploaded has some border mismatches, see below.
> 
> This will be an issue when using borders to do things like auto-define transmission links. It may be a gadm/nuts combo issue.

@brynpickering that looks nasty... and it is odd because I directly used the SCEC / EC files!
Is this both `ehighways` and `eu+`?

brynpickering commented:
> Is this both ehighways and eu+?

Just checked. Yes, both.

brynpickering commented:
It's very likely due to the gadm/nuts mismatch as we use GADM at the national and ehighways resolutions for: Albania, Bosnia and Herzegovina, North Macedonia, Montenegro, Serbia.

The problem goes away for the regional resolution, where we use almost entirely GADM

irm-codebase commented:
Even the regional resolution has problems (also due to GDAM / NUTS mismatches).
I created an issue on euro-calliope to track this. I'll look for alternatives for `clio` in the meantime

irm-codebase commented:
@brynpickering about the metadata.
It seems like pandas already saves metadata in `df.attrs` when you save to parquet. Unfortunately, geopandas does not support this for some reason (see here https://github.com/geopandas/geopandas/issues/3320).

For now, we should add attributions as a separate column. Using `pyarrow` would add too much overhead.

irm-codebase commented:
@brynpickering I updated the `mne` and `prt` files with simplification and a column stating their source.

I also used this opportunity to test how metadata can be embedded in `.parquet` files. There are some limitations, but it's still relatively easy.

- `pandas` will serialise any values in `df.attrs`, and recover them when re-loading `.parquet`.
- `geopandas` lacks this functionality at the moment, but it may be within [their near-term scope](https://github.com/geopandas/geopandas/issues/3320).

`attrs` only works at file-level. Per-column attrs is not a thing for now.
I updated our docs, telling people to save metadata this way, and requiring a bare minimum of 3 values (units, source, license).

brynpickering commented on docs/development_guide/requirements.md:
````diff
@@ -41,17 +41,37 @@ The following is a list of general advice on how to format files to help tools i
+            ```
 
 3. **Raster data**: we prefer to use GeoTIFF (.tiff) files.
 4. **Polygon data**: we prefer [GeoParquet](https://geoparquet.org/) (.parquet) files.
````
Add an `example` admonition here and point to the shape resources we provide? You could use the `pymdownx.snippets` mkdocs extension to get text directly from `resources/shapes/readme.md` (`<!--- --8<-- [start:docs] -->` and `<!--- --8<-- [end:docs] -->` in the README and `--8<-- "resources/shapes/README.md:docs"` in this file).

irm-codebase replied:
Easy enough addition!

brynpickering reviewed (approved):
Suggestion for injecting an example in the docs, but not a _must_ to merge this in!
